packages/langium/test/lsp/find-references.test.ts

```diff
@@ -0,0 +1,505 @@
+/******************************************************************************
+ * Copyright 2021 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+
+import { createLangiumGrammarServices } from '../../src';
+import { expectFindReferences } from '../../src/test';
+
+const grammarServices = createLangiumGrammarServices().grammar;
+const findReferences = expectFindReferences(grammarServices);
+
+describe('findReferences', () => {
+    test('Must find references to parent interface in another interface declaration. Including parent interface declaration', async () => {
+        const grammar = `grammar test hidden (WS)
+        terminal ID: /\\w+/;
+
+        interface <|<|>A|> {
```

Comment:
Much better to read. Thank you!

One thing that is not clear on first sight is, what is meant? Like "AHA! There is an index and two ranges. So, it might be the case that the test picks out the index and points to that position to find all references which are equal to the ranges."...

For the moment it is good like this, I believe. The only thing we could do better is giving these ranges and indices names as a kind of handles.

But as I said, it is already a huge improvement